repository_service_tuf_worker/repository.py

```diff
@@ -1212,3 +1308,91 @@ def metadata_rotation(
         logging.warn(deprecation_message)
 
         return self.metadata_update(payload, update_state)
+
+    def _sign_root(
+        self, root: Metadata[Root], sig_payload: Dict
+    ) -> Tuple[bool, str]:
+        """
+        Sign the root metadata and call actions depending on the bootstrap
+        state, and if it matches the threshold
+        """
+        signature: Signature = Signature.from_dict(sig_payload)
+        root.signatures[signature.keyid] = signature
+
+        signed = self._validate_signatures(root, Root.type)
+        bootstrap = self._settings.get_fresh("BOOTSTRAP")
+
+        if "signing-" in bootstrap:
```

Comment:
Might be worth to check this a lot earlier, before you deserialise the metadata, or the signature, or perform anything crypto verification.